ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/Transaction.java

```diff
@@ -426,9 +429,9 @@ public String toString() {
 
     protected Address sender;
 
-    protected int chainId = -1;
+    protected BigInteger chainId = BigInteger.valueOf(-1);
 
-    public Builder chainId(final int chainId) {
+    public Builder chainId(final BigInteger chainId) {
       this.chainId = chainId;
```

Comment:
Two possible approaches here, either `public Builder chainId(final Optional<BigInteger> chainId)` which is just directly assigned or `public Builder chainId(final BigInteger chainId)` which is wrapped with `Optional.of` - if you don't have a chain ID don't call `chainId` on the builder.

Reply:
done. used approach of defaulting to Optional.empty only calling chainId on builder if you have a chainId